bin/certbot-auth.sh.aws

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+set -e
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd "$DIR/.."
+
+[[ -z "$CERTBOT_VALIDATION" ]] && { echo "error: this script is designed to run through 'certbot ... --manual-auth-hook=$0' (\$CERTBOT_VALIDATION is not set)."; exit -1; }
+[[ -z "$CERTBOT_DOMAIN" ]] && { echo "error: this script is designed to run through 'certbot ... --manual-auth-hook=$0' (\$CERTBOT_DOMAIN is not set)."; exit -1; }
+
+# CERTBOT_DOMAIN = aas.dirac.institute
+HOST=_acme-challenge.$(echo $CERTBOT_DOMAIN | cut -d . -f 1)
+DOMAIN=$(echo $CERTBOT_DOMAIN | cut -d . -f 2-)
+
+hosted_zone_id=$(aws --output=json route53 list-hosted-zones | jq --arg DOMAIN "${DOMAIN}." ' .HostedZones | .[] | select(.Name==$DOMAIN) | .Id ')
+# remove trailing / preceeding quotes
+hosted_zone_id="${hosted_zone_id%\"}"
+hosted_zone_id="${hosted_zone_id#\"}"
+
+record_dns_name="${HOST}.${DOMAIN}"
+
+resources=$(aws --output=json route53 list-resource-record-sets --hosted-zone-id $hosted_zone_id)
+host_txt_record=$(echo $resources | jq --arg record_dns_name "${record_dns_name}." ' .ResourceRecordSets | .[] | select(.Name==$record_dns_name and .Type=="TXT")')
+
+if [[ $host_txt_record ]]; then
+    name=$(echo $host_txt_record | jq '.Name')
+    type=$(echo $host_txt_record | jq '.Type')
+    ttl=$(echo $host_txt_record | jq '.TTL')
+    resource_records=$(echo $host_txt_record | jq '.ResourceRecords')
+
+    delete_request=$(cat <<EOF
```

Comment:
While it doesn't really matter (since JSON is insensitive to whitespaces), you could also start the heredoc with `<<-EOF`). The `<<-FOO` syntax allows you to indent the text (but must be with tabs!), and bash will automatically remove the (common) indentation. See https://www.oreilly.com/library/view/bash-cookbook/0596526784/ch03s04.html for more.

Reply:
done